internal/k8s/ako_init.go

```diff
@@ -121,6 +121,93 @@ func deleteConfigFromConfigmap(cs kubernetes.Interface) bool {
 	return true
 }
 
+func (c *AviController) SetSEGroupCloudName() bool {
+	seGroupToUse := lib.GetSEGNameEnv()
+	client := avicache.SharedAVIClients().AviClient[0]
+	var err error
+	// 2. Marker based (only advancedL4)
+	if seGroupToUse == "" && lib.GetAdvancedL4() {
+		err, seGroupToUse = lib.FetchSEGroupWithMarkerSet(client)
```

Comment:
In `FetchSEGroupWithMarkerSet`, we have reference to utils.CloudName. But We are setting `Cloudname` either in ako-infra code base or at line number: 150. I think this will result in failure for advanceL4 deployment.

Reply:
I think similar comment is true for function `validateAndConfigureSeGroup`

Reply:
This is interesting. We were never setting cloud name before. init() function gets called automatically which sets the cloud name from env variable. So for normal AKO this works fine.  
In case of VCF we don't have the cloud name in env, so it has to be set explicitly.
Removed SetCloudName for non vcf setup.